rocAL/source/pipeline/master_graph.cpp

```diff
@@ -1131,15 +1131,22 @@ std::vector<rocalTensorList *> MasterGraph::create_tf_record_meta_data_reader(co
     return _metadata_output_tensor_list;
 }
 
-std::vector<rocalTensorList *> MasterGraph::create_label_reader(const char *source_path, MetaDataReaderType reader_type) {
+std::vector<rocalTensorList *> MasterGraph::create_label_reader(const char *source_path, MetaDataReaderType reader_type, const char *file_list_path) {
     if (_meta_data_reader)
         THROW("A metadata reader has already been created")
     if (_augmented_meta_data)
         THROW("Metadata can only have a single output")
-
-    MetaDataConfig config(MetaDataType::Label, reader_type, source_path);
+    if (strlen(source_path) == 0 && strlen(file_list_path) == 0)
+        THROW("Either source path or file list path needs to be provided")
+    const char *root_path;
+    if (strlen(file_list_path) == 0) {
```

Comment:
checking for nullptr is better

Reply:
We are setting "" as default value for char arrays so nullptr check won't work here